biggerquery/dagbuilder.py

```diff
@@ -0,0 +1,127 @@
+
+    dag_deployment_id = _get_dag_deployment_id(workflow.workflow_id, start_from, build_ver)
+    dag_file_path = get_dags_output_dir(workdir) / (dag_deployment_id + '_dag.py')
+
+    TIMEZONE_OFFSET_SECONDS = str(datetime.now().astimezone().tzinfo.utcoffset(None).seconds)
+    if workflow.schedule_interval == '@daily':
+        start_from = start_from[:10]
+        start_date_as_str = f'datetime.strptime("{start_from}", "%Y-%m-%d")'
+    else:
+        start_date_as_str = f'datetime.strptime("{start_from}", "%Y-%m-%d %H:%M:%S") - (timedelta(seconds={TIMEZONE_OFFSET_SECONDS}))'
+
+    print(f'dag_file_path: {dag_file_path.resolve()}')
+
+    dag_chunks = []
+
+    dag_chunks.append("""    
+from airflow import DAG
+from datetime import timedelta
+from datetime import datetime
+from airflow.contrib.operators import kubernetes_pod_operator
+
+default_args = {{
+            'owner': 'airflow',
+            'depends_on_past': True,
+            'start_date': {start_date_as_str},
+            'email_on_failure': False,
+            'email_on_retry': False,
+            'execution_timeout': timedelta(minutes=90)
+}}
+
+dag = DAG(
+    '{dag_id}',
+    default_args=default_args,
+    max_active_runs=1,
+    schedule_interval='{schedule_interval}'
+)
+""".format(dag_id=dag_deployment_id,
+           start_date_as_str=start_date_as_str,
+           schedule_interval=workflow.schedule_interval),
+
+)
+
+    def get_job(workflow_job):
+        return workflow_job.job
+
+    def build_dag_operator(workflow_job, dependencies):
+        job = get_job(workflow_job)
+        job_var = "t" + str(job.id)
+        task_id = job.id.replace("_","-")
+
+        dag_chunks.append("""
+{job_var} = kubernetes_pod_operator.KubernetesPodOperator(
+    task_id='{task_id}',
+    name='{task_id}',
+    cmds=['bgq'],
+    arguments=['run', '--job', '{bgq_job}', '--runtime', '{{{{ execution_date.strftime("%Y-%m-%d %H:%M:%S") }}}}', '--root', '{root_folder}', '--config', '{{{{var.value.env}}}}'],
+    namespace='default',
+    image='{docker_image}',
+    is_delete_operator_pod=True,
+    retries={retries},
+    retry_delay= timedelta(seconds={retry_delay}),
+    dag=dag)            
+""".format(job_var=job_var,
+          task_id=task_id,
+          docker_image = docker_repository+":"+build_ver,
+          bgq_job= workflow.workflow_id+"."+job.id,
+          root_folder=root_package_name,
+          retries=job.retry_count,
+          retry_delay=job.retry_pause_sec)
+)
+
+        for d in dependencies:
+            up_job_var = "t" + str(get_job(d).id)
+            dag_chunks.append("{job_var}.set_upstream({up_job_var})".format(job_var=job_var, up_job_var=up_job_var))
+
+
+    workflow.call_on_graph_nodes(build_dag_operator)
+
+    dag_file_content = '\n'.join(dag_chunks) + '\n'
+    print (dag_file_content)
+    dag_file_path.write_text(dag_file_content)
+
+    return dag_file_path.as_posix()
+
+
+def _get_dag_deployment_id(workflow_name: str,
```

Comment:
I think we don't need to make this function "private". I suggest that we use a single strategy for making things private, that is:

Make things private only in the classes and modules that are exported as a user interface.